### PR TITLE
Cursed guns patch

### PR DESCRIPTION
**Pull request by FluffyTowels**
## Additions

Compatability for all guns of the Cursed Guns mod as well as the fedora hat from it.

## Reasoning
 
- I tried to make the guns based on stats of similar guns that already have integrated patches.
- I felt that having a separate 40mm launcher to launch only rubber bullets was unnecessary, so I removed it in favor of adding a 40mm rubber slug that all 40mm grenade launchers should be able to use.
- The fedora hat was patched in exactly the same way as the cowboy hat because I think they're similar enough.
- The 40mm rubber slugs use the 40mm Grenade HE sprites for ammo, because making an ammo sprite that would fit the existing aesthetic is too hard for me.
- The Ballistic Knife Blades ammuthing for the kniper use the dart sprites for ammo, because making an ammo sprite that would fit the existing aesthetic is too hard for me.

## Alternatives

- Patch High Impact Dongopod to be the only gun that is able to use 40mm Rubber slugs

## Testing

- [X] Game runs without errors with and without patched mod loaded (At least any that were related to the patch or CE)
- [X] Playtested a colony (Around 5 hours while testing if every gun works)
- [X] Tested every gun and new ammo types and they seem to work without errors

## Notes

- There may be some problems I haven't noticed due to my inexperience. 
- I am willing to consider any changes to the patch for the sake of it better fitting in with the rest.

### CODE REVIEW

File: Patches/Cursed Guns/Fedora.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		<value>
+			<StuffEffectMultiplierArmor>2</StuffEffectMultiplierArmor>
+		</value>
+	</li>
+  
+	<li Class="PatchOperationAdd">
+		<xpath>Defs/ThingDef[defName="Apparel_FedoraCursedGun"]/apparel</xpath>
+		<value>
+			<layers>
+				<li>MiddleHead</li>
```
Comment (N7Huntsman) [inline Patches/Cursed Guns/Fedora.xml:22]: The MiddleHead layer has been done away with, so this whole operation can just be removed.

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
Thanks, I hate it. It appears _adequate_. I won't call it "fine" because there there is nothing "fine" about this...

In all seriousness, though, the patch itself seems to be good to go, save an operation w/ the fedora that can be removed. The rubber grenade works and all the weapons look like they've been patched.

File: Patches/Cursed Guns/Fedora.xml (hunk 1 above)
Comment (FluffyTowels) [reply Patches/Cursed Guns/Fedora.xml:22]: I'm still new to this kind of thing, so I'm not sure, if you're asking me to do something. Or how to do it.

**Comment by FluffyTowels** [comment]
I think I removed the operation from the fedora correctly, but I'm not sure.

**Review by N7Huntsman — APPROVED** [review]

**Pull request merged**